mesh_database_client/database.py

```diff
@@ -18,37 +18,53 @@ class DatabaseClient:
 
     def __init__(self, spreadsheet_id = None):
 
-        if spreadsheet_id is None:
+        self.spreadsheet_id = spreadsheet_id
+        if self.spreadsheet_id is None:
             self.spreadsheet_id = os.environ.get("SPREADSHEET_ID")
-        else:
-            self.spreadsheet_id = spreadsheet_id
 
         SCOPES = ['https://www.googleapis.com/auth/spreadsheets.readonly']
         # created in Google Cloud admin
         SERVICE_ACCOUNT_FILE = 'credentials.json'
 
-        signup_sheet_range = 'Form Responses 1!A:AO'
-        links_sheet_range = 'Links!A:I'
-
-        CREDS = service_account.Credentials.from_service_account_file(
-        SERVICE_ACCOUNT_FILE, scopes=SCOPES)
-
-        self.service = build('sheets', 'v4', credentials=CREDS)
+        credentials_dict = self._get_sheets_credentials_from_env()
+        credentials = service_account.Credentials.from_service_account_info(credentials_dict)
+        self.service = build('sheets', 'v4', credentials=credentials)
 
         # Call the Sheets API
         self.sheet = self.service.spreadsheets()
 
-        # load all data into dataframe
-        self.signup_df = self.get_range_as_df(signup_sheet_range)
-        self.links_df = self.get_range_as_df(links_sheet_range)
+        self.signup_df = self.get_signup_df()
+        self.links_df = self.get_links_df()
 
-        self.process_signup_df()
-        self.process_links_df()
+        self.gmaps = googlemaps.Client(key=os.environ.get("MAPS_API"))
+
+    def _get_sheets_credentials_from_env(self):
+        try:
+            credentials_mapping = {
+                "type": "GOOGLE_SHEETS_TYPE",
+                "project_id": "GOOGLE_SHEETS_PROJECT_ID",
+                "private_key_id": "GOOGLE_SHEETS_PRIVATE_KEY_ID",
+                "private_key": "GOOGLE_SHEETS_PRIVATE_KEY",
+                "client_email": "GOOGLE_SHEETS_CLIEND_EMAIL",
+                "client_id": "GOOGLE_SHEETS_CLIENT_ID",
+                "auth_uri": "GOOGLE_SHEETS_AUTH_URI",
+                "token_uri": "GOOGLE_SHEETS_TOKEN_URI",
+                "auth_provider_x509_cert_url" : "GOOGLE_SHEETS_AUTH_PROVIDER_X509_CERT_URL",
+                "client_x509_cert_url" : "GOOGLE_SHEETS_CLIENT_509_CERT_URL"
```

Comment:
Can we also update the dotenv template with these variables?

Reply:
added them